### PR TITLE
ASSIMP file import

### PR DESCRIPTION
**Pull request by eero-lehtinen**
Works for singular meshes.
Doesn't take into account materials or textures.

I can't get windows to build, but add assimp to vcpkg just in case.

### CODE REVIEW

File: test/monkey.obj
Comment (juliusikkala) [inline test/monkey.obj:1]: I would prefer that these testing .obj files wouldn't be in the repo. `test.glb` should be enough, I didn't intend it to be a file format test but rendering tests instead. The reason I'm pretty strict about this is that I have seen too many 500mb repos where it's all just oversized test data. The `test.obj` here is 211k lines of diff as well :D

File: src/obj.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        mesh_object& obj = add_unique_named(obj_name, md.mesh_objects);
+        obj.set_model(&md.models[name]);
+    }
+
+    for(auto& m: md.meshes)
+        m->refresh_buffers();
+
+    md.directional_lights["default"] = directional_light(vec3(-0.5, -0.5, -0.5));
+
+    md.cameras["default"] = camera();
```
Comment (juliusikkala) [inline src/obj.cc:157]: No need, there should be a default camera already without this.

**Review by juliusikkala — COMMENTED** [review]
It seems to be on the right track, especially once the materials start working. Because the implementation is done using ASSIMP, I don't think there's a reason for the loader to be specific to .obj, right? Maybe this could be a generic fallback loader whenever the format is not .ply or .glb.

File: test/monkey.obj
Comment (eero-lehtinen) [reply test/monkey.obj:1]: I force pushed a new commit without these so they won't bloat git history.

File: src/assimp.cc [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+                ai_mesh->mNormals[j].z
+            );
+        
+        if(ai_mesh->HasTangentsAndBitangents())
+            v.tangent = vec4(
+                ai_mesh->mTangents[j].x,
+                ai_mesh->mTangents[j].y,
+                ai_mesh->mTangents[j].z,
+                // Not sure if this is correct
+                1.0f
```
Comment (juliusikkala) [inline src/assimp.cc:69]: It's indeed not, and the correct value can be calculated from the mBitangents member. See https://github.com/vga-group/tauray/blob/ad3773041877e16be9d8157b1ad973d50e1fc98d/shader/forward.vert#L32 for how it's used. 
I think the correct value here would be `sign(dot(ai_mesh->mBitangents[j], cross(ai_mesh->mNormals[j], ai_mesh->mTangents[j])))`.

**Review by juliusikkala — CHANGES_REQUESTED** [review]
Seems to be mostly fine, for an experimental importer for now. I'll probably add animations + do something about deducing the metallic and roughness factors from phong parameters at some point later on.

Once the tangent thing is fixed, I think this is ready for merging from my point of view.

**Pull request merged**